File: InsuraTech.Domain/Events/PolicyActivatedEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using InsuraTech.Domain.Common;

namespace InsuraTech.Domain.Events
{
    public sealed class PolicyActivatedEvent : IDomainEvent
    {


        public Guid EventId { get; } = Guid.NewGuid();
        public Guid Id => EventId;

        public DateTime OccurredOn { get; } = DateTime.UtcNow;
        public string EventType => nameof(PolicyActivatedEvent);


        public Guid PolicyId           {  get; }
        public string PolicyNumber     {  get; }
        public string InsuredFullNamed {  get; }
        public string InsuredDocumentId { get; }

        public PolicyActivatedEvent(Guid policyId, string policyNumber, string insuredFullNamed, string insuredDocumentId)
        {
            PolicyId = policyId;
            PolicyNumber = policyNumber;
            InsuredFullNamed = insuredFullNamed;
            InsuredDocumentId = insuredDocumentId;
        }

    }
}

File: backend_v2/src/main/java/com/insuratech/domain/events/PolicyActivatedEvent.java
package com.insuratech.domain.events;

import com.insuratech.domain.common.IDomainEvent;

import java.time.Instant;

public record PolicyActivatedEvent(String policyId, String policyNumber, Instant occurredOn) implements IDomainEvent {
    public PolicyActivatedEvent(String policyId, String policyNumber) {
        this(policyId, policyNumber, Instant.now());
    }
}
